src/Query.tsx

```diff
@@ -358,30 +358,46 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
 
   private getQueryResult = (): QueryResult<TData, TVariables> => {
     let data = { data: Object.create(null) as TData } as any;
-    // attach bound methods
+    // Attach bound methods
     Object.assign(data, observableQueryFields(this.queryObservable!));
-    // fetch the current result (if any) from the store
-    const currentResult = this.queryObservable!.currentResult();
-    const { loading, networkStatus, errors } = currentResult;
-    let { error } = currentResult;
-    // until a set naming convention for networkError and graphQLErrors is decided upon, we map errors (graphQLErrors) to the error props
-    if (errors && errors.length > 0) {
-      error = new ApolloError({ graphQLErrors: errors });
-    }
-
-    Object.assign(data, { loading, networkStatus, error });
 
-    if (loading) {
-      Object.assign(data.data, this.previousData, currentResult.data);
-    } else if (error) {
-      Object.assign(data, {
-        data: (this.queryObservable!.getLastResult() || {}).data,
-      });
+    // When skipping a query (ie. we're not querying for data but still want
+    // to render children), make sure the `data` is cleared out and
+    // `loading` is set to `false` (since we aren't loading anything).
+    if (this.props.skip) {
```

Comment:
Quick note here - I've re-arranged this a bit to short circuit un-necessary extra query logic, when the query is to be skipped. One thing to add though, is that this means we're leaving `networkStatus` as `undefined`. This is currently intentional, as setting this to `NetworkStatus.ready` can cause problems when using something like the  `graphql` hoc and calculating the `skip` value using a function (forcing a `NetworkStatus.ready` value can prematurely end rendering). Leaving `networkStatus` as `undefined` here seems to be okay though, and is better than the way it was before when we did the `skip` check at the end of `getQueryResult`. That left `networkStatus` in `NetworkStatus.loading` state, even though we weren't loading anything.